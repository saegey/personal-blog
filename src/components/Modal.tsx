import { Text, Flex, Box, Close } from 'theme-ui'

type ModalProps = {
  children: JSX.Element
  modalOpen: (arg: boolean) => void
  headerText: string
}

const Modal = ({ children, modalOpen, headerText }: ModalProps) => {
  return (
    <Box
      title="modalBackgroundContainer"
      variant="styles.faded"
      onClick={() => {
        modalOpen(false)
      }}
    >
      <Box
        sx={{
          backgroundColor: 'background',
          margin: ['0% auto', '5% auto', '5% auto'],
          padding: '0px',
          width: '100%',
          maxWidth: '706px',
          overflowY: 'scroll',
        }}
        onClick={event => {
          event.stopPropagation()
        }}
      >
        <Flex
          sx={{
            position: 'sticky',
            top: '0px',
            backgroundColor: 'background',
          }}
        >
          <Box sx={{ p: '20px' }}>
            <Text as="h2" variant="resultsHeading">
              {headerText}
            </Text>
          </Box>
          <Box sx={{ marginLeft: 'auto', p: '20px' }}>
            <Close
              title="Close"
              ml="auto"
              mr={-2}
              sx={{ color: 'text', cursor: 'pointer' }}
              onClick={() => {
                modalOpen(false)
              }}
            />
          </Box>
        </Flex>
        <Box sx={{ paddingX: '20px' }}>{children}</Box>
      </Box>
    </Box>
  )
}

export default Modal
