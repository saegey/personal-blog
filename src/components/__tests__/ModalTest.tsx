/**
 * @jest-environment jsdom
 */
import renderer from 'react-test-renderer'
import { fireEvent, render, screen } from '@testing-library/react'
import { ThemeProvider } from 'theme-ui'

import Modal from '../Modal'

import { theme } from './__test-utils__'

describe('Modal', () => {
  it('renders', () => {
    const json = renderer
      .create(
        <Modal modalOpen={jest.fn()} headerText={'hello world'}>
          <h1>tst</h1>
        </Modal>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  it('clicking close button close runs function', async () => {
    const modalClose = jest.fn()
    render(
      <Modal modalOpen={modalClose} headerText={'hello world'}>
        <h1>tst</h1>
      </Modal>
    )
    const closeButton = await screen.findByTitle('Close')
    fireEvent.click(closeButton)
    expect(modalClose).toBeCalledTimes(1)
  })

  it('clicking outside modal runs close function', async () => {
    const modalClose = jest.fn()
    render(
      <Modal modalOpen={modalClose} headerText={'hello world'}>
        <h1>tst</h1>
      </Modal>
    )
    // screen.debug()
    const modalBackground = await screen.findByTitle('modalBackgroundContainer')
    fireEvent.click(modalBackground)
    expect(modalClose).toBeCalledTimes(1)
  })
})
