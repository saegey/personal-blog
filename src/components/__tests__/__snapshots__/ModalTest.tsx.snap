// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: background;
  margin: 0% auto;
  padding: 0px;
  width: 100%;
  max-width: 706px;
  overflow-y: scroll;
}

@media screen and (min-width: 40em) {
  .emotion-1 {
    margin: 5% auto;
  }
}

@media screen and (min-width: 52em) {
  .emotion-1 {
    margin: 5% auto;
  }
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: -webkit-sticky;
  position: sticky;
  top: 0px;
  background-color: background;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 20px;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  padding: 20px;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 4px;
  width: 32px;
  height: 32px;
  color: inherit;
  background-color: transparent;
  border: none;
  border-radius: 4px;
  margin-right: -8px;
  margin-left: auto;
  color: text;
  cursor: pointer;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 20px;
  padding-right: 20px;
}

<div
  className="emotion-0"
  onClick={[Function]}
  title="modalBackgroundContainer"
>
  <div
    className="emotion-1"
    onClick={[Function]}
  >
    <div
      className="emotion-2"
    >
      <div
        className="emotion-3"
      >
        <h2
          className="emotion-0"
        >
          hello world
        </h2>
      </div>
      <div
        className="emotion-5"
      >
        <button
          aria-label="Close"
          className="emotion-6"
          onClick={[Function]}
          title="Close"
        >
          <svg
            fill="currentColor"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="emotion-7"
    >
      <h1>
        tst
      </h1>
    </div>
  </div>
</div>
`;
